Automattic/wp-calypso
# Jetpack Backup, Jetpack Scan: Show VaultPress experience for sites with reason=host_not_supported

A continuation (in spirit) of #46167. When a site's Rewind status is `unavailable` with a reason of `host_not_supported`, this implies the site is using VaultPress, and should also be covered by the `HasVaultPressSwitch` component.

#### Changes proposed in this Pull Request

* Add `host_not_supported` to the list of reasons that cause `HasVaultPressSwitch` to evaluate to its true condition.
* Add a test to cover this new condition.
* Optimize function calls a bit by using `useCallback`.
* Convert `getRewindState` to use TypeScript, to get rid of errors in other TypeScript files that depend on it.

#### Testing instructions

Testing procedures are the same as **Scenario 1** in #46167, with the exception that the site you're testing must meet all the following criteria:
1. Must use VaultPress instead of Rewind;
2. Must not be hosted by Pressable; and
3. Must not be owned by an Automattician. (If you're an Automattician, consider using an alternate account to set up your test site.)

I've copied the instructions here for easy access:

* In **both** Jetpack Cloud and Calypso, select a site that uses VaultPress for backups and scanning.
* Navigate to the Backup and Scan pages.
* Verify that on both pages, you see messaging that says your site uses VaultPress, and a button that redirects you to your site's VaultPress dashboard.

<img width="382" alt="Screen Shot 2020-10-06 at 07 25 46" src="https://user-images.githubusercontent.com/670067/95201414-88242300-07a5-11eb-9dc1-75a62996fe39.png"> <img width="383" alt="Screen Shot 2020-10-06 at 07 25 35" src="https://user-images.githubusercontent.com/670067/95201431-8e1a0400-07a5-11eb-93c7-0a9e44f79cb6.png">

<img width="384" alt="Screen Shot 2020-10-06 at 07 27 25" src="https://user-images.githubusercontent.com/670067/95201505-ac7fff80-07a5-11eb-815d-774c3503c075.png"> <img width="385" alt="Screen Shot 2020-10-06 at 07 27 35" src="https://user-images.githubusercontent.com/670067/95201525-b0ac1d00-07a5-11eb-9ea4-6ed15ab2da5e.png">
